partner_email_check/models/res_partner.py

```diff
@@ -9,30 +9,84 @@
 _logger = logging.getLogger(__name__)
 
 try:
-    from validate_email import validate_email
+    from email_validator import validate_email, EmailNotValidError
 except ImportError:
-    _logger.debug('Cannot import "validate_email".')
+    _logger.debug('Cannot import "email_validator".')
 
-    def validate_email(email):
-        _logger.warning(
-            'Can not validate email, '
-            'python dependency required "validate_email"')
-        return True
+    validate_email = None
 
 
 class ResPartner(models.Model):
     _inherit = 'res.partner'
 
-    @api.constrains('email')
-    def constrains_email(self):
-        for rec in self.filtered("email"):
-            self.email_check(rec.email)
-
     @api.model
     def email_check(self, emails):
-        for email in emails.split(','):
-            if not validate_email(email):
-                raise ValidationError(
-                    _("%s is an invalid email") % email.strip()
-                )
-        return True
+        return ','.join(self._normalize_email(email.strip())
+                        for email in emails.split(','))
+
+    def _normalize_email(self, email):
+        if validate_email is None:
+            _logger.warning(
+                'Can not validate email, '
+                'python dependency required "email_validator"')
+            return email
+
+        try:
+            result = validate_email(
+                email,
+                check_deliverability=self._should_check_deliverability(),
+            )
+        except EmailNotValidError as e:
+            raise ValidationError(
+                _("%s is an invalid email: %s") % (email.strip(), e.message)
```

Comment:
how does `email_validator` handle translations? I doubt we can make this pick up Odoo's translation mechanism easily, so instead of having a mixed localized/non-localized string, I think we'll be better off not using its messages at all